catalog/dags/data_augmentation/rekognition/add_rekognition_labels_dag.py

```diff
@@ -0,0 +1,132 @@
+the DAG from a specific position in the file if it fails.
+
+This DAG is idempotent and can be run multiple times without issue.
+"""
+
+import logging
+from datetime import timedelta
+
+from airflow.decorators import dag
+from airflow.operators.trigger_dagrun import TriggerDagRunOperator
+
+from common.constants import DAG_DEFAULT_ARGS, POSTGRES_CONN_ID
+from common.slack import notify_slack
+from common.sql import run_sql
+from data_augmentation.rekognition import constants
+from data_augmentation.rekognition.add_rekognition_labels import (
+    notify_parse_complete,
+    parse_and_insert_labels,
+    resume_insertion,
+)
+from database.batched_update.constants import DAG_ID as BATCHED_UPDATE_DAG_ID
+
+
+logger = logging.getLogger(__name__)
+
+
+@dag(
+    dag_id=constants.DAG_ID,
+    schedule=None,
+    catchup=False,
+    tags=["data_augmentation"],
+    doc_md=__doc__,
+    default_args={
+        **DAG_DEFAULT_ARGS,
+        "retries": 0,
+        "execution_timeout": timedelta(hours=5),
+    },
+    render_template_as_native_obj=True,
+)
+def add_rekognition_labels():
+    check_for_resume = resume_insertion()
+
+    notify_start = notify_slack.override(task_id=constants.NOTIFY_START_TASK_ID)(
+        text=f"Starting Rekognition label insertion\n"
+        f"{constants.TEMPLATE_SLACK_MESSAGE_CONFIG}",
+        dag_id=constants.DAG_ID,
+        username=constants.SLACK_USERNAME,
+        icon_emoji=constants.SLACK_ICON,
+    )
+
+    notify_resume = notify_slack.override(
+        task_id=constants.NOTIFY_RESUME_TASK_ID
+    )(
+        text="Resuming Rekognition label insertion "  # noqa: UP031
+        "from position: `{{ var.value.%s }}`\n%s"
+        % (constants.CURRENT_POS_VAR_NAME, constants.TEMPLATE_SLACK_MESSAGE_CONFIG),
+        dag_id=constants.DAG_ID,
+        username=constants.SLACK_USERNAME,
+        icon_emoji=constants.SLACK_ICON,
+    )
+
+    create_temp_table = run_sql.override(
+        task_id="create_loading_table",
+        execution_timeout=timedelta(minutes=1),
+    )(sql_template=constants.CREATE_TEMP_TABLE_QUERY)
+
+    create_temp_table_index = run_sql.override(
+        task_id="create_temp_table_index",
+        execution_timeout=timedelta(minutes=1),
+    )(sql_template=constants.CREATE_TEMP_TABLE_INDEX_QUERY)
+
+    # These values are interpolated to allow overriding locally while still
+    # defining a default value in an obvious location
+    insert_labels = parse_and_insert_labels(
+        s3_bucket=constants.S3_BUCKET,
+        s3_prefix=constants.TEMPLATE_S3_PREFIX,
+        in_memory_buffer_size=constants.TEMPLATE_IN_MEMORY_BUFFER_SIZE,
+        file_buffer_size=constants.TEMPLATE_FILE_BUFFER_SIZE,
+        postgres_conn_id=POSTGRES_CONN_ID,
+    )
+
+    notify_parse_complete(insert_labels)
+
+    batched_update = TriggerDagRunOperator(
+        task_id="trigger_batched_update",
+        trigger_dag_id=BATCHED_UPDATE_DAG_ID,
```

Comment:
Should we be triggering the `automated_batched_update` instead since this is automated? I don't have a strong opinion either way 😄

Reply:
Since this is a _one-off_ import, that's sort of why I was sending it to the the regular `batched_update` 😅 If this were an ingestion we were doing regularly, I'd definitely have this be the `automated_batched_update`!

Reply:
Makes sense to me!